core/src/main/java/com/datastax/oss/driver/internal/core/config/cloud/CloudConfigFactory.java

```diff
@@ -228,12 +230,22 @@ protected TrustManagerFactory createTrustManagerFactory(
   @NonNull
   protected BufferedReader fetchProxyMetadata(
       @NonNull URL metadataServiceUrl, @NonNull SSLContext sslContext) throws IOException {
-    HttpsURLConnection connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
-    connection.setSSLSocketFactory(sslContext.getSocketFactory());
-    connection.setRequestMethod("GET");
-    connection.setRequestProperty("host", "localhost");
-    return new BufferedReader(
-        new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+    try {
+      HttpsURLConnection connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
+      connection.setSSLSocketFactory(sslContext.getSocketFactory());
+      connection.setRequestMethod("GET");
+      connection.setRequestProperty("host", "localhost");
+      return new BufferedReader(
+          new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+    } catch (ConnectException e) {
```

Comment:
In theory we should have a test for this in `CloudConfigFactoryTest` but I guess it will be complex to emulate a `ConnectException`...

Reply:
It would be difficult and I think more importantly of somewhat limited utility.